hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -897,7 +897,8 @@ object HoodieSparkSqlWriter {
     // for backward compatibility
     if (hiveSyncEnabled) {
       metaSyncEnabled = true
-      syncClientToolClassSet += classOf[HiveSyncTool].getName
+      if (syncClientToolClassSet.isEmpty)
```

Comment:
when would the set be empty?

Reply:
You are right, it has a default value. Then will just keep the default.